how-it-works/1_protocol/01_peer-to-peer-p2p.subpage.md

```diff
@@ -5,10 +5,73 @@ shareImage: /img/how-it-works/peer-to-peer-p2p.600.jpg
 slug: peer-to-peer-p2p
 ---
 
-# Peer-to-Peer Layer
+# Peer-to-Peer
 
-[![Watch youtube video](https://i.ytimg.com/vi/HOQb0lKIy9I/maxresdefault.jpg)](https://www.youtube.com/watch?v=HOQb0lKIy9I)
+The peer-to-peer layer (P2P) of the Internet Computer realizes the reliable and secure communication of *network messages*, also called *artifacts*, between the nodes of a subnet.
+Artifacts are the network messages that are to be broadcast in the subnet, such as ingress messages to canister smart contracts submitted by users or protocol-originating messages such as signature shares computed by the consensus layer of subnet nodes.
+P2P realizes a peer-to-peer broadcast network that guarantees the secure eventual broadcast delivery of an artifact.
+The P2P layer thereby is the communication fabric for the IC protocol stack and is used by the consensus layer, the next layer in the stack above it, to broadcast artifacts to the nodes in the subnet.
+
+## Gossip Protocol
+
+The P2P layer builds upon the basic principle of *gossip*.
+Gossip in communication networks works along the same basic principle as gossip among people: A node in the subnet is connected with a subset of the other nodes of the subnet – its *peers*.
+Whenever a node receives a message from a peer or creates one itself as part of the IC protocol, it gossips this message to all its peers.
+By every node in the network doing exactly this, every artifact *eventually* propagates through the whole subnet.
+
+## Asynchronous Communication Networks
```

Comment:
Why does this come up here? Reliable broadcast doesn't really seem harder in the asynchronous setting, but consensus certainly is. Shall we maybe move it there?

Reply:
Agreed